components/board/client.go

```diff
@@ -232,6 +249,29 @@ func (arc *analogReaderClient) Read(ctx context.Context, extra map[string]interf
 	return int(resp.Value), nil
 }
 
+// analogWriterClient satisfies a gRPC based board.AnalogWriter. Refer to the interface
+// for descriptions of its methods.
+type analogWriterClient struct {
+	*client
+	boardName string
```

Comment:
board should not be here right?

I want to stop following the pattern for other things in the board  implementation - can we have the write method more closely follow the API, like in other components? If there are helper functions needed, let's make them unexported on board.

Reply:
I thought we were following how GPIO and PWM are doing it, where the proto requests have `name` which is the board name and `pin` which is pinName.  Should we not do that?

Reply:
wdym by what is exposed? all the api functions in board have either `name` or `board_name` in the request which both represent the board name. If we don't need the board name in the request we could just have it take the pin name and value but then I would need to change the current analog writer API to remove `name`.

Reply:
or should we have `name` be pinName and `pin` be the number?

Reply:
What are we doing in the board drivers? How are we using that name?

when I use the SDK to get a board, then a gpio_pin_by_name, then setting that pin to high or low - what name am I giving the gpio_pin_by_name, I don't think it's the board's name.

Reply:
when calling gpio_pin_by_name we are giving the pin name. But no in the driver you would define an `AnalogWriter` struct with the board and pin, and then call `AnalogWriter.Write(ctx, value, extra)`

Reply:
This may be a case where we just have to clean up board proto, but I am really wondering what is useful. 

Look at everything Pete is return nils  from [here](https://github.com/viam-labs/viam-revolution-pi/blob/main/revolution_pi/revolution_pi.go) - have we overloaded our interface with helper functions and can we avoid that further. I think we do need a helper to get the analog reader by name or number form the board, but I expect that function signature to be more like: pi.AnalogWrite("11", 255)

Reply:
Don't worry about removing other functions from the board interface - let's just keep this modular board implementation in mind when cleaning - I want to make the new API as simple and clean and "not gross" as possible, so yes if AnalogWrite("pin", value) is what's useful - let's do that.

Since it's in the API here's what we'll do - email the  a removal of the board attribute, say we don't actually need it and it's just adding extra helper functions for nothing - wait for any objections for a bit, but let's go ahead and make a pr with the change in API.

Reply:
I do not think this requires a change to the API - even if we get rid of the AnalogWriter interface we would still need board in the API request so we can specify what board we want to write to.

Reply:
I was suggesting we could get rid of the AnalogWriter interface in `board.go` and instead do the server side implementation the way GPIO and PWM is doing it. But this would not remove the requirement for boardName.

Reply:
you're right - i'm mixing up the go and SDK interface with the API rpc. Thank you for figuring that out and discussing it. So, yes, let's simplify `AnalgoWrite` to directly write to the pin - which is what we do with pin.Set and PWM eventually in all the drivers.

Reply:
Somewhere between the code in this file and getting the RDK to do stuff is a GRPC request, and that must contain the board name. This file (and the Go SDK more broadly) seem overly complicated, but I think we'll always need some way of specifying the board name in the GRPC request. (The way I'd simplify this is to have the board client contain a `WriteAnalog` method, rather than an `AnalogWriterByName` which returns a separate object that contains a `Write` method. but either way, the client needs to know the board's name when it sends the request)

Reply:
^ yep we figured that out, thanks Alan.

Reply:
(sorry, Github didn't show me the latest updates)